### PR TITLE
Adds flag optional params to `IssuedCurrencyClient::createOffer`

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

The `CreateOffer` transaction type has flag options that can be applied to the resulting offers. This PR adds those as optional parameters to the `createOffer` method in `IssuedCurrencyClient`.

### Context of Change

IOUs

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Before / After

`IssuedCurrencyClient::createOffer` now has additional optional parameters for the flags

## Test Plan

Added tests for the flags, including invalid combinations. CI passes.

### CODE REVIEW

File: src/XRP/issued-currency-client.ts [hunk 1]
```diff
@@ -1222,13 +1223,30 @@ export default class IssuedCurrencyClient {
    *               It is not considered an error if the offer specified does not exist.
    * @param expiration (Optional) Time after which the offer is no longer active, in seconds since the Ripple Epoch.
    *               (See https://xrpl.org/basic-data-types.html#specifying-time)
+   * @param passive (Optional) If enabled, the offer does not consume offers that exactly match it, and instead becomes
```
Comment (amiecorso) [inline src/XRP/issued-currency-client.ts:1226]: I wonder if we should specify "Defaults to false." for all of these opt flag params?

File: test/XRP/issued-currency-client-integration.test.ts [hunk 2]
```diff
@@ -873,6 +873,98 @@ describe('IssuedCurrencyClient Integration Tests', function (): void {
...
+
+    const offerSequenceNumber = 1
+
+    const rippleEpochStartTimeSeconds = 946684800
+    const currentTimeUnixEpochSeconds = Date.now() / 1000 // 1000 ms/sec
+    const currentTimeRippleEpochSeconds =
+      currentTimeUnixEpochSeconds - rippleEpochStartTimeSeconds
+    const expiration = currentTimeRippleEpochSeconds + 60 * 60 // roughly one hour in future
+
+    // WHEN the wallet creates an offer to exchange (receive) their own issued currency for their XRP (deliver)
```
Comment (amiecorso) [inline test/XRP/issued-currency-client-integration.test.ts:948]: make a note about the flags we're testing here in the WHEN statement?  it seems that immediateOrCancel and fillOrKill are mutually exclusive? but it's not noted here

File: src/XRP/shared/offer-create-flags.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+ * Flags used in OfferCreate transactions.
+ *
+ * @see https://xrpl.org/offercreate.html#offercreate-flags
+ */
+class OfferCreateFlags {
+  static TF_PASSIVE = 0x00010000
+  static TF_IMMEDIATE_OR_CANCEL = 0x00020000
+  static TF_FILL_OR_KILL = 0x00040000
+  static TF_SELL = 0x00080000
+}
```
Comment (tedkalaw) [inline src/XRP/shared/offer-create-flags.ts:11]: any reason to not use an enum for these instead? they end up being implemented pretty much the same under the covers - objects with named properties

**Review by tedkalaw — APPROVED** [review]

**Review by amiecorso — APPROVED** [review]
just a few notes!

**Pull request merged**